src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -368,22 +408,76 @@
+        <foreach item="studyId" collection="studyViewFilterHelper.studyViewFilter.studyIds" open="(" separator="," close=")">
+            #{studyId}
+
+        </foreach>
+    </sql>
+    
     <sql id="categoricalClinicalDataCountFilter">
-        SELECT ${unique_id}
-        FROM ${table_name}
-        WHERE attribute_name = '${clinicalDataFilter.attributeId}' AND
-        type='${type}'
-        <foreach item="dataFilterValue" collection="clinicalDataFilter.values" open=" AND ((" separator=") OR (" close="))">
-            <trim prefix="" prefixOverrides="AND">
-                AND (
+        <!-- check if 'NA' is selected -->
+        <bind name="userSelectsNA" value="false" />
+        <bind name="userSelectsValue" value="false" />
+        <foreach item="dataFilterValue" collection="clinicalDataFilter.values">
+            <choose>
+                <when test="dataFilterValue.value == 'NA'">
+                    <bind name="userSelectsNA" value="true" />
+                </when>
+                <otherwise>
+                    <bind name="userSelectsValue" value="true" />
+                </otherwise>
+            </choose>
+        </foreach>
+        (
+        <!-- if 'NA' is selected, prepare NA samples -->
+        <if test="userSelectsNA">
+            SELECT DISTINCT ${unique_id}
+            FROM sample_derived sd
+            LEFT JOIN (<include refid="selectAllClinicalDataByAttribute"/>) AS categorical_clinical_data
+            ON
+            <choose>
+                <when test="'${type}' == 'sample'">
+                    sd.sample_unique_id = categorical_clinical_data.sample_unique_id
+                </when>
+                <otherwise>
+                    sd.patient_unique_id = categorical_clinical_data.patient_unique_id
+                </otherwise>
+            </choose>
+            WHERE empty(attribute_value)
+            AND EXISTS (<include refid="selectAllClinicalDataByAttribute"/>)
+        </if>
+
+        <!-- if both 'NA' and non-NA are selected, union them together -->
+        <if test="userSelectsNA and userSelectsValue">
+            UNION ALL
+        </if>
+
+        <!-- if non-NA is selected, prepare non-NA samples -->
+        <if test="userSelectsValue">
+            SELECT ${unique_id}
+            FROM ${table_name}
+            WHERE attribute_name = '${clinicalDataFilter.attributeId}' AND
+            type='${type}'
+            <foreach item="dataFilterValue" collection="clinicalDataFilter.values" open=" AND ((" separator=") OR (" close="))">
+                <trim prefix="" prefixOverrides="AND">
+                    AND (
                     <include refid="normalizeAttributeValue">
                         <property name="attribute_value" value="attribute_value"/>
                     </include>
-                ) ILIKE '${dataFilterValue.value}'
-            </trim>
-        </foreach>
+                    ) ILIKE '${dataFilterValue.value}'
+                </trim>
+            </foreach>
```

Comment:
From above mentioned comment you can bring the 470-478 up into the first NA part